src/web_client.h

```diff
@@ -18,73 +18,76 @@ extern int web_enable_gzip, web_gzip_level, web_gzip_strategy, web_donotrack_com
 #define COOKIE_MAX 1024
 #define ORIGIN_MAX 1024
 
+/** Web client response message. */
 struct response {
-    BUFFER *header;                 // our response header
-    BUFFER *header_output;          // internal use
-    BUFFER *data;                   // our response data buffer
+    BUFFER *header;                 ///< our response header
+    BUFFER *header_output;          ///< internal use
+    BUFFER *data;                   ///< our response data buffer
 
-    int code;                       // the HTTP response code
+    int code;                       ///< the HTTP response code
 
-    size_t rlen;                    // if non-zero, the excepted size of ifd (input of firecopy)
-    size_t sent;                    // current data length sent to output
+    size_t rlen;                    ///< if non-zero, the excepted size of ifd (input of firecopy)
+    size_t sent;                    ///< current data length sent to output
 
-    int zoutput;                    // if set to 1, web_client_send() will send compressed data
+    int zoutput;                    ///< if set to 1, web_client_send() will send compressed data
 #ifdef NETDATA_WITH_ZLIB
-    z_stream zstream;               // zlib stream for sending compressed output to client
-    Bytef zbuffer[ZLIB_CHUNK];      // temporary buffer for storing compressed output
-    size_t zsent;                   // the compressed bytes we have sent to the client
-    size_t zhave;                   // the compressed bytes that we have received from zlib
-    int zinitialized:1;
+    z_stream zstream;               ///< zlib stream for sending compressed output to client
+    Bytef zbuffer[ZLIB_CHUNK];      ///< temporary buffer for storing compressed output
+    size_t zsent;                   ///< the compressed bytes we have sent to the client
+    size_t zhave;                   ///< the compressed bytes that we have received from zlib
+    int zinitialized:1;             ///< ktsaour: Your help needed
```

Comment:
netdata supports delayed initialization of the compression library for each request. This keeps track if the compression library has been initialized or not.
to understand it better, focus on the single threaded web server: everything is async, so we need to keep states in variables.